tools/src/main/java/io/narayana/db/Allocator.java

```diff
@@ -103,6 +115,93 @@ static boolean waitForTcp(final String host, final int port, final int connTimeo
         return false;
     }
 
+    /**
+     * Loads proper driver dynamically.
+     * Credit: http://www.kfu.com/%7Ensayer/Java/dyn-jdbc.html
+     * <p>
+     * Executes a simple SELECT 1 statement to verify the DB actually does something.
+     *
+     * @param dbUrl            JDBC connection URL
+     * @param user             DB user
+     * @param pass             DB password
+     * @param driverClass      e.g. org.postgresql.Driver
+     * @param pathToDriverJar  Either a downloaded driver jar or a one from ShrinkWrap Maven resolver
+     * @param overallTimeoutMs How long do we keep trying with 1000ms pause in between
+     * @return true on success
+     * @throws ClassNotFoundException Fix the driver class name
+     * @throws MalformedURLException  Fix the driver jar path
+     */
+    public static boolean executeTestStatement(
+            final String dbUrl, final String user, final String pass, final String driverClass, final String pathToDriverJar, final long overallTimeoutMs)
+            throws ClassNotFoundException, MalformedURLException, IllegalAccessException, InstantiationException, SQLException {
+
+        final String sql = "SELECT 1;";
```

Comment:
Is it possible to make this sql configurable ? It could be useful if we need to support more DB types in the future.